java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/expr/EvaluatorVisitor.java

```diff
@@ -4531,7 +4531,7 @@ private static ReferenceType getOrLoadClass(VirtualMachine vm, String name,
             String message = Bundle.MSG_IncompatibleThreadStateMessage();
             InvalidExpressionException ieex = new InvalidExpressionException(message, itsex);
             throw new IllegalStateException(ieex);
-        } catch (Exception ex) {
+        } catch (ClassNotLoadedException | InvalidTypeException | InvocationException ex) {
             return null;
```

Comment:
this is risky. I probably would recommend to revert this. Unless we test this a bit more.

For example the line above throws `IllegalStateException`, which would have been caught before with `Exception` and null would have been returned, But now this wouldn't be caught, it would skip to the finally block and then just propagate the exception further.

I don't know what the original intention was. Maybe it was supposed to propagate, maybe not. So you either fixed a bug here or added one :) In any case it would be safer to not hide this in a cleanup commit. Someone familiar with the debugger should review this part.

Reply:
done..